### PR TITLE
feat(cluster-dialer): cluster dialer support pipeline-agent register

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
cluster dialer support pipeline-agent register

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda-org.erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJhc3NpZ25lZSI6WyIxMDAxMjA1Il19&id=304696&iterationID=1174&pId=301655&type=TASK)

#### Specified Reviewers:

/assign @your-reviewer

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |   cluster dialer support pipeline-agent register           |
| 🇨🇳 中文    |     cluster-dialer支持pipeline客户端的注册         |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: apistructs/cluster_dialer.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+)
+
+func (c ClusterDialerHeaderKey) String() string {
+	return string(c)
+}
+
+type ClusterDialerClusterType string
+
+var (
+	ClusterDialerClusterTypePipeline ClusterDialerClusterType = "pipeline"
```
Comment (iutx) [inline apistructs/cluster_dialer.go:35]: Pipeline does not belong to cluster, rename to dialer agent type better.

Comment (chengjoey) [reply apistructs/cluster_dialer.go]: adjusted, ClusterDialerClusterTypePipeline -> ClusterDialerClusterComponentTypePipeline
type ClusterDialerClusterType rename to ClusterDialerClusterComponentType

File: apistructs/cluster_dialer.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	ClusterDialerHeaderKeyClusterComponentType ClusterDialerHeaderKey = "X-Erda-Cluster-Component-Type"
+	ClusterDialerHeaderKeyClusterInfo          ClusterDialerHeaderKey = "X-Erda-Cluster-Info"
+	ClusterDialerHeaderKeyAuthorization        ClusterDialerHeaderKey = "Authorization"
+)
+
+func (c ClusterDialerHeaderKey) String() string {
+	return string(c)
+}
+
+type ClusterDialerClusterComponentType string
```
Comment (sfwn) [inline apistructs/cluster_dialer.go:32]: ClusterDialerKeyType

ClusterDialerKeyTypeDefault = "" (cluster)
ClusterDialerKeyTypePipeline = "pipeline"

Comment (sfwn) [inline apistructs/cluster_dialer.go:32]: ClusterDialerClientType

ClusterDialerClientTypeDefault = "" (cluster)
ClusterDialerClientTypeCluster = "cluster" (cluster)
ClusterDialerClientTypePipeline = "pipeline"

File: apistructs/cluster_dialer.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+var (
+	ClusterDialerClusterComponentTypePipeline ClusterDialerClusterComponentType = "pipeline"
+)
+
+func (c ClusterDialerClusterComponentType) String() string {
+	return string(c)
+}
+
+func (c ClusterDialerClusterComponentType) MakeClientKeyByClusterKey(clusterKey string) string {
```
Comment (sfwn) [inline apistructs/cluster_dialer.go:42]: strange.

File: bundle/cluster_dialer.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+package bundle
+
+import (
+	"encoding/json"
+
+	"github.com/erda-project/erda/bundle/apierrors"
+)
+
+func (b *Bundle) IsClusterExistedInDialer(clusterKey, clusterType string) (bool, error) {
```
Comment (sfwn) [inline bundle/cluster_dialer.go:23]: IsClusterDialerClientRegistered(clientType, clusterKey)

"", ${clusterKey}
pipeline, ${clusterKey}

File: modules/cluster-dialer/auth/auth.go [hunk 5]
```diff
@@ -70,8 +70,9 @@ func (a *Authorizer) Authorizer(req *http.Request) (string, bool, error) {
 		return "proxy", true, nil
 	}
 
-	clusterKey := req.Header.Get("X-Erda-Cluster-Key")
-	authInfo := req.Header.Get("Authorization")
+	clusterKey := req.Header.Get(apistructs.ClusterDialerHeaderKeyClusterKey.String())
+	authInfo := req.Header.Get(apistructs.ClusterDialerHeaderKeyAuthorization.String())
+	clusterType := apistructs.ClusterDialerClusterComponentType(req.Header.Get(apistructs.ClusterDialerHeaderKeyClusterComponentType.String()))
```
Comment (sfwn) [inline modules/cluster-dialer/auth/auth.go:75]: clientType

**Review by sfwn — CHANGES_REQUESTED** [review]

File: apistructs/cluster_dialer.go [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+	ClusterDialerClientTypeDefault  ClusterDialerClientType = ""         // cluster
+	ClusterDialerClientTypeCluster  ClusterDialerClientType = "cluster"  // cluster
+	ClusterDialerClientTypePipeline ClusterDialerClientType = "pipeline" // pipeline
+)
+
+func (c ClusterDialerClientType) String() string {
+	return string(c)
+}
+
+func (c ClusterDialerClientType) MakeClientKey(clusterKey string) string {
```
Comment (sfwn) [inline apistructs/cluster_dialer.go:44]: MakeClientKeyByClusterKey

**Review by sfwn — APPROVED** [review]

**Pull request merged**